source/core/GenericFunctions.cpp

```diff
@@ -228,6 +233,10 @@ InstructionCore* EmitGenericBinOpInstruction(ClumpParseState* pInstructionBuilde
     if (savedOperation.CompareCStr("Split") || savedOperation.CompareCStr("Join")) {  // Split and Join are uniquely identified by source type rather than dest type
         goalType = sourceXType;
     }
+    if (argCount == 4 &&
```

Comment:
Consider moving validation code early in the method. It seems we are doing unnecessary calculations if we are going to exit early. Also I was wondering about the "pInstructionBuilder->_argTypes[2]->BitEncoding()!=destType->BitEncoding()" check, since destType is actually pInstructionBuilder->_argTypes[2], but you fixed this is a later change.